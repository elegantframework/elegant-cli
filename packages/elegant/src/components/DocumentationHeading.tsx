export default function DocumentationHeading({
    title,
    section = ""
}: {
    title: string;
    section?: string;
}){
    return(
    <div className="relative z-20">
        <div>
            {section && (
                <p className="mb-2 text-sm leading-6 font-semibold text-primary-500 dark:text-primary-400">
                    {section}
                </p>
            )}
            <div className="flex items-center">
                <h1 className="inline-block text-2xl sm:text-3xl font-extrabold text-slate-900 tracking-tight dark:text-slate-200">
                    {title}
                </h1>
            </div>
        </div>
    </div>
    );
}