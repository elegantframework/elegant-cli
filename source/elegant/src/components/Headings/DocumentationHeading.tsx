type Props = {
    /**
     * The title of the heading element.
     */
    title: string;
    /**
     * The navigation section that this element is located in.
     */
    section?: string;
};

/**
 * A heading component to be used within the documentation layout component.
 * @returns An html heading section to be displayed at the top of documentation pages.
 */
export function DocumentationHeading({
    title,
    section = ""
}: Props){
    return(
    <div className="relative z-20">
        <div>
            {section && (
                <p className="mb-2 text-sm leading-6 font-semibold text-primary-500 dark:text-primary-400">
                    {section}
                </p>
            )}
            <div className="flex items-center">
                <h1 className="inline-block text-2xl sm:text-3xl font-extrabold text-slate-900 tracking-tight dark:text-slate-200">
                    {title}
                </h1>
            </div>
        </div>
    </div>
    );
}