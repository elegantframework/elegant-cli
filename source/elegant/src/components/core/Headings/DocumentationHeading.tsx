type Props = {
    title: string;
    description: string;
    section?: string;
};

/**
 * 
 * @returns 
 */
export function DocumentationHeading({
    title,
    description,
    section = ""
}: Props){
    return(
    <div className="relative z-20">
        <div>
            {section && (
                <p className="mb-2 text-sm leading-6 font-semibold text-primary-500 dark:text-primary-400">
                    {section}
                </p>
            )}
            <div className="flex items-center">
                <h1 className="inline-block text-2xl sm:text-3xl font-extrabold text-slate-900 tracking-tight dark:text-slate-200">
                    {title}
                </h1>
            </div>
        </div>
        {description && (
            <p className="mt-2 text-lg text-slate-700 dark:text-slate-400">
                {description}
            </p>
        )}
    </div>
    );
}