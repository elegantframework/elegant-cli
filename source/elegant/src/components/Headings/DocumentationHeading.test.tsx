import React from "react";
import renderer from 'react-test-renderer';
import { DocumentationHeading } from "./DocumentationHeading";

describe("Documentation Heading component", () => {
    it('renders the documentation heading properly', () => {
        const heading = renderer
        .create(
            <DocumentationHeading 
                title="Unit Test"
                section="Hello world"  
            />
        )
        .toJSON();
        expect(heading).toMatchSnapshot();
    });
});