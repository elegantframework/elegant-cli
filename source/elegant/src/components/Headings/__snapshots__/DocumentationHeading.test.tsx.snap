// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Documentation Heading component renders the documentation heading properly 1`] = `
<div
  className="relative z-20"
>
  <div>
    <p
      className="mb-2 text-sm leading-6 font-semibold text-primary-500 dark:text-primary-400"
    >
      Hello world
    </p>
    <div
      className="flex items-center"
    >
      <h1
        className="inline-block text-2xl sm:text-3xl font-extrabold text-slate-900 tracking-tight dark:text-slate-200"
      >
        Unit Test
      </h1>
    </div>
  </div>
</div>
`;
